components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/CurrencyInputView.kt

```diff
@@ -21,49 +21,65 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
 
 @Composable
 fun CurrencyInputView(
     modifier: Modifier = Modifier,
+    initialInputValue: String = "",
     onInputValueChange: ((String) -> Unit)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
     placeholderText: String? = null,
     errorText: String? = null,
     errorContentDescription: String? = null,
-    labelText: String? = null,
-    labelContentDescription: String? = null,
-    singleLine: Boolean = false,
-    characterCount: Int? = null,
+    singleLine: Boolean = true,
     enableDecimal: Boolean = true,
 ) {
 
     // pattern matches a decimal number
-    val decimalPattern = remember { Regex("^(?!0[0-9])[0-9]+(\\.?)([0-9]?[0-9]?)") }
+    val decimalPattern = remember { Regex("^([0-9]*)(\\.?)([0-9]*)$") }
```

Comment:
Jemma, is it allowed?
<img width="441" alt="Screenshot 2024-01-10 at 11 08 21" src="https://github.com/hmrc/android-components/assets/11981999/75473c6d-845c-47bf-ab19-2ff8be3dbbf3">

Reply:
I debated this for a while but I think it's okay at the component level. The component accepts any decimal number and then it will be down to the consumer to validate if it's an acceptable input.